docs/content/app-protect/configuration.md

```diff
@@ -203,3 +203,173 @@ spec:
   softwareVersion: 15.1.0
   tag: Fruits
 ```
+
+## OpenAPI Specification in NGINX Ingress Controller
+
+The OpenAPI Specification defines the spec file format needed to describe RESTful APIs. The spec file can be written either in JSON or YAML. Using a spec file simplifies the work of implementing API protection. Refer to the [OpenAPI Specification](#https://github.com/OAI/OpenAPI-Specification) (formerly called Swagger) for details. 
+
+NGINX Ingress Controller supports OpenAPI Specification versions 2.0 and 3.0.
+
+The simplest way to create an API protection policy is using an OpenAPI Specification file to import the details of the APIs. If you use an OpenAPI Specification file, NGINX App Protect WAF will automatically create a policy for the following properties (depending on what's included in the spec file):
+* Methods
+* URLs
+* Parameters
+* JSON profiles 
+
+An OpenAPI-ready policy template is provided with the NGINX App Protect WAF packages and is located in: `/etc/app_protect/conf/NginxApiSecurityPolicy.json`
+
+It contains violations related to OpenAPI set to blocking (enforced).
+
+### Types of OpenAPI References
+
+There are different ways of referencing OpenAPI Specification files. The configuration is similar to [External References](#external-references).
+
+**Note**: Any update of an OpenAPI Specification file referenced in the policy will not trigger a policy compilation. This action needs to be done actively by reloading the NGINX configuration.
+
+#### URL Reference
+
+URL reference is the method of referencing an external source by providing its full URL.
+
+Make sure to configure certificates prior to using the HTTPS protocol - see the [External References](#https-reference) for details.
+
+## Configuration in NGINX Ingress Controller
+
+These are the typical steps to deploy an OpenAPI protection Policy in NGINX Ingress Controller:
+
+1. Copy the API security policy `/etc/app_protect/conf/NginxApiSecurityPolicy.json` to a different file so that it can be edited.
+2. Add the reference to the desired OpenAPI file.
+3. Make other custom changes if needed (e.g. enable Data Guard protection).
+4. Use a tool to convert the result to YAML. There are many, for example: [`yq` utility](https://github.com/mikefarah/yq).
+5. Add the YAML properties to create an `APPolicy` Custom Resource putting the policy itself (as in step 4) within the `spec` property of the Custom Resource.
+6. Refer to [App Protect Policies](#app-protect-policies) section above.
+7. Add an annotation referencing that `APPolicy` resource and apply that resource. See the same section above.
```

Comment:
Thinking about this - @brianehlert in your opinion, should we be avoiding using the term "annotation" because it has special meaning in Kubernetes, however irrelevant in this context?

Reply:
To your point - 'annotation' does mean something specific in K8s.
It is specific to the Ingress resource.
If the step is specific to the Ingress resource referencing the policy that is valid.

But then that also means the example is specific to using the Ingress resource and we need to also document the VirtualService resource.

Reply:
So, shouldn't the steps above go further into the configuration? Add a step to create a `Policy` object, then attach that `Policy`object to a `VirtualServer` resource?

Reply:
Any walkthrough should be 'complete'
There might be a step that forks and then comes back together.  But that should be called out.

Reply:
@aknot242 where do we add this additional step in the above sequence? 

create a Policy object, then attach that Policyobject to a VirtualServer resource

Reply:
@rranghar I am thinking the additional step(s) would replace the current step 7, as the current step it in itself does not go into attaching the `APPolicy` to a `Policy` resource, then attaching that policy resource to a `VirtualServer` resource.

Reply:
sure, replaced step 7 with additional info. as below:

7. Create a `Policy` object, then attach that `Policy` object to a `VirtualServer` resource.